### ISSUE TITLE
Fatal error on latest PHP 8.x releases

### ISSUE THREAD

**Issue by jonasraoni** [pkp/pkp-lib#9643]
**Describe the bug**
More details at the "how to reproduce" section.
The code `(new IntlDateFormatter('C', 0, 0))->format(new DateTime())` generates a fatal error on the latest PHP 8.x releases, see a [reproduction case here](https://onlinephp.io?s=s7EvyCjg5UpNzshX0MhLLVfwzCvJcUksSXXLL8pNLClJLdJQd1bX1NS1SwMLgNWA5EMyc1M1NDWtAQ%2C%2C&v=8.3.1%2C8.2.14%2C8.1.27%2C7.4.33) and the exception below.

```php
PHP Fatal error:  Uncaught Error: Found unconstructed IntlDateFormatter in /var/www/html/lib/pkp/lib/vendor/php81_bc/strftime/src/php-8.1-strftime.php:101
Stack trace:
#0 /var/www/html/lib/pkp/lib/vendor/php81_bc/strftime/src/php-8.1-strftime.php(101): IntlDateFormatter->format(Object(DateTime))
#1 /var/www/html/lib/pkp/lib/vendor/php81_bc/strftime/src/php-8.1-strftime.php(202): PHP81_BC\\{closure}(Object(DateTime), '%B')
#2 [internal function]: PHP81_BC\\{closure}(Array)
#3 /var/www/html/lib/pkp/lib/vendor/php81_bc/strftime/src/php-8.1-strftime.php(181): preg_replace_callback('/(?<!%)%([_#-]?...', Object(Closure), '%B')
#4 /var/www/html/lib/pkp/classes/template/PKPTemplateManager.inc.php(2266): PHP81_BC\\strftime('%B', Object(DateTime))
#5 /var/www/html/cache/t_compile/91c4852bbf0ab9d68a26a0a5d3411137ee5758f8^756a7119cb48731a8fd45c250b45f4bcca2d6b01_0.app.frontendpagessearch.tpl.php(88): PKPTemplateManager->smartyHtmlSelectDateA11y(Array, Object(Smarty_Internal_Template))
#6 /var/www/html/lib/pkp/lib/vendor/smarty/smarty/libs/sysplugins/smarty_template_resource_base.php(123): content_64c2d4d46b0704_54914599(Object(Smarty_Internal_Template))
#7 /var/www/html/lib/pkp/lib/vendor/smarty/smarty/libs/sysplugins/smarty_template_compiled.php(114): Smarty_Template_Resource_Base->getRenderedTemplateCode(Object(Smarty_Internal_Template))
#8 /var/www/html/lib/pkp/lib/vendor/smarty/smarty/libs/sysplugins/smarty_internal_template.php(217): Smarty_Template_Compiled->render(Object(Smarty_Internal_Template))
#9 /var/www/html/lib/pkp/lib/vendor/smarty/smarty/libs/sysplugins/smarty_internal_templatebase.php(238): Smarty_Internal_Template->render(false, 1)
#10 /var/www/html/lib/pkp/lib/vendor/smarty/smarty/libs/sysplugins/smarty_internal_templatebase.php(134): Smarty_Internal_TemplateBase->_execute(Object(Smarty_Internal_Template), NULL, '91c4852bbf0ab9d...', NULL, 1)
#11 /var/www/html/lib/pkp/classes/template/PKPTemplateManager.inc.php(1207): Smarty_Internal_TemplateBase->display('frontend/pages/...', NULL, '91c4852bbf0ab9d...', NULL)
#12 /var/www/html/pages/search/SearchHandler.inc.php(153): PKPTemplateManager->display('frontend/pages/...')
#13 /var/www/html/pages/search/SearchHandler.inc.php(38): SearchHandler->search(Array, Object(Request))
#14 /var/www/html/lib/pkp/classes/core/PKPRouter.inc.php(397): SearchHandler->index(Array, Object(Request))
#15 /var/www/html/lib/pkp/classes/core/PKPPageRouter.inc.php(246): PKPRouter->_authorizeInitializeAndCallRequest(Array, Object(Request), Array, false)
#16 /var/www/html/lib/pkp/classes/core/Dispatcher.inc.php(144): PKPPageRouter->route(Object(Request))
#17 /var/www/html/lib/pkp/classes/core/PKPApplication.inc.php(374): Dispatcher->dispatch(Object(Request))
#18 /var/www/html/index.php(68): PKPApplication->execute()
#19 {main}
  thrown in /var/www/html/lib/pkp/lib/vendor/php81_bc/strftime/src/php-8.1-strftime.php on line 101
```

**Solution**
I've filed an issue at upstream, but I can't see an easy way to solve it, so I'm not going to contribute with a fix: https://github.com/alphp/strftime/issues/17

On our side, we can add a custom patch for the library, which would be basically to add one extra line to the top of the function:
`$locale = $locale ?? \AppLocale::getLocale();`

Even if there's no Exception, passing `C` (or the last `setlocale()` call that worked) as the locale is not going to give us the expected behavior.

**To Reproduce**
1. Use a latest PHP 8.x release
2. Use a locale which is available in OJS, but not in the system (i.e. `locale -a`), then the `setlocale()` call will fail and the default locale will remain (`C`).
3. Access the search at the frontend
4. See a fatal error

**What application are you using?**
OJS 3.3

**Comment by jonasraoni** [comment]
@asmecher Let me know if you agree with the fix
- https://github.com/pkp/pkp-lib/pull/9647

**Comment by asmecher** [comment]
Hmm, perhaps an OK way out of a bad situation. I'm OK to merge this until we get an upstream fix -- either for https://github.com/alphp/strftime/issues/17 or for https://github.com/smarty-php/smarty/issues/810.

**Comment by jonasraoni** [comment]
The exception was fixed in [PHP 8.2.15](https://github.com/php/php-src/issues/12943).
Anyway, I'll merge the patch, because the way they retrieve the locale (i.e. `setlocale(LC_TIME, '0')`) isn't enough for probably a lot of users (e.g. if you do a `setlocale(LC_ALL, 'pt_BR')`, and the `pt_BR` doesn't exist in the `locale -a`, the locale will not be changed).

We do have this issue in the PS.
